habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/exec/UvCommandHelper.java

```
@@ -0,0 +1,180 @@
+            Pattern versionPattern = Pattern.compile(extractUvVersionRegex);
+            Matcher matchedPattern = versionPattern.matcher(versionResult); 
+            if (matchedPattern.find()){
+                String version = matchedPattern.group(1);
+                return new ImmutablePair<Boolean, String>(true, version); 
+            } else {
+                throw new HabushuException("uv version pattern not found.");
+            }
+            
+        } catch (Throwable e) {
+            return new ImmutablePair<Boolean, String>(false, null);
+        }
+    }
+
+    /**
+     * Retrieves the version of Python that is set for the configured working
+     * directory via the .python-version file.
+     *
+     * @return
+     */
+    public String getCurrentPythonVersion() throws MojoExecutionException {
+        String currentPythonVersion = pythonVersionFileContents();
+
+        if (!currentPythonVersion.isEmpty()) {
+            return currentPythonVersion;
+        } else throw new MojoExecutionException("A .python-version file is required to have a reproducible build. To create this file, from the package directory, run `uv python pin <python-version>`.");
+    }
+
+    /**
+     * Updates the .python-version file to reflect the desired Python version.
+     *
+     * @return
+     */
+    public void updatePythonVersion(String targetVersion) throws MojoExecutionException {
+        List<String> pythonPinCommand = createPythonPinCommand(targetVersion);
+        sharedCommandHelper.execute(pythonPinCommand);
+    }
+
+    /**
+     * Returns whether the specified dependency package is installed within this
+     * uv project's virtual environment (and pyproject.toml).
+     *
+     * @param packageName
+     * @return
+     */
+    public boolean isDependencyInstalled(String packageName) {
+        try {
+            sharedCommandHelper.execute(Arrays.asList("pip", "show", packageName));
+        } catch (Throwable e) {
+            return false;
```

S: If this results in an exception, probably something exceptional happened that should cause us to fail hard?

---

might also be good to provide error message as well.

---

This method is used to check if `behave`, `black` etc. are installed. If false is returned, the user is notified in the next step of the pipeline that it is being installed.